data/org.eclipse.birt.report.engine.script.javascript/src/org/eclipse/birt/report/engine/javascript/ScriptUtil.java

```diff
@@ -22,14 +22,31 @@
 import org.mozilla.javascript.PolicySecurityController;
 import org.mozilla.javascript.SecurityController;
 
+/**
+ * Script util class
+ *
+ * @since 3.3
+ *
+ */
 public class ScriptUtil {
 
+	/**
+	 * Create security controller
+	 *
+	 * @return the created security controller
+	 */
 	public static SecurityController createSecurityController() {
 		return new PolicySecurityController();
 	}
 
+	/**
+	 * Get the security domain
+	 *
+	 * @param file JavaScript file
+	 * @return the secure JavaScript code
+	 */
 	public static Object getSecurityDomain(final String file) {
-		if ((file == null) || (System.getSecurityManager() == null)) {
+		if ((file == null) || !(new JavascriptVersion()).isECMAScriptSecurityEnabled()) {
```

Comment:
I do not know enough about the SecurityManager nor the deprecation of it, but as I understand it, this is a way to give Rhino "what it wants" without the presence of a SecurityManager. Would not that help in cases like #1780 ?

Reply:
Changed to static access.